services/src/main/java/edu/unc/lib/dl/cdr/services/rest/modify/ExportCsvController.java

```diff
@@ -0,0 +1,261 @@
+import edu.unc.lib.dl.search.solr.model.SearchRequest;
+import edu.unc.lib.dl.search.solr.model.SearchResultResponse;
+import edu.unc.lib.dl.search.solr.model.SearchState;
+import edu.unc.lib.dl.search.solr.service.ChildrenCountService;
+import edu.unc.lib.dl.search.solr.util.FacetConstants;
+import edu.unc.lib.dl.search.solr.util.SearchFieldKeys;
+import edu.unc.lib.dl.ui.controller.AbstractSolrSearchController;
+
+/**
+ * Controller which generates a CSV listing of a repository object
+ * and all of its children, recursively depth first.
+ *
+ * @author bbpennel
+ */
+@Controller
+@RequestMapping("exportTree/csv")
+public class ExportCsvController extends AbstractSolrSearchController {
+    private static final Logger log = LoggerFactory.getLogger(ExportCsvController.class);
+
+    public static final String OBJ_TYPE_HEADER = "Object Type";
+    public static final String PID_HEADER = "PID";
+    public static final String TITLE_HEADER = "Title";
+    public static final String PATH_HEADER = "Path";
+    public static final String LABEL_HEADER = "Label";
+    public static final String DEPTH_HEADER = "Depth";
+    public static final String DELETED_HEADER = "Deleted";
+    public static final String DATE_ADDED_HEADER = "Date Added";
+    public static final String DATE_UPDATED_HEADER = "Date Updated";
+    public static final String MIME_TYPE_HEADER = "MIME Type";
+    public static final String CHECKSUM_HEADER = "Checksum";
+    public static final String FILE_SIZE_HEADER = "File Size (bytes)";
+    public static final String NUM_CHILDREN_HEADER = "Number of Children";
+    public static final String DESCRIBED_HEADER = "Description";
+
+    private static final String[] CSV_HEADERS = new String[] {
+            OBJ_TYPE_HEADER, PID_HEADER, TITLE_HEADER, PATH_HEADER, LABEL_HEADER,
+            DEPTH_HEADER, DELETED_HEADER, DATE_ADDED_HEADER, DATE_UPDATED_HEADER,
+            MIME_TYPE_HEADER, CHECKSUM_HEADER, FILE_SIZE_HEADER, NUM_CHILDREN_HEADER,
+            DESCRIBED_HEADER};
+
+    @Autowired
+    private ChildrenCountService childrenCountService;
+
+    protected SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd kk:mm:ss");
+
+    @RequestMapping(value = "{pid}", method = RequestMethod.GET)
+    public ResponseEntity<Map<String, Object>> export(@PathVariable("pid") String pidString, HttpServletRequest request,
+                                                      HttpServletResponse response) throws IOException {
+
+        Map<String, Object> result = new HashMap<>();
+        result.put("action", "exportCsv");
+        result.put("username", request.getRemoteUser());
+
+        try {
+            if (pidString.equals("collections")) {
+                throw new FedoraException("Not allowed to download repository root");
+            }
+
+            PID pid = PIDs.get(pidString);
+
+            SearchRequest searchRequest = generateSearchRequest(request, searchStateFactory.createSearchState());
```

Comment:
Could you add in an access control check before this step to verify that the user has `viewHidden` to the requested resource before pulling back solr results? As a side note, I can't remember if checking that will throw a NotFoundException if it can't find the resource.